### ISSUE TITLE
ChipSet componentWillUnmount is called before this.state.foundation is set

### ISSUE THREAD

**Issue by dugusword** [material-components/material-components-web-react#675]
this.setState({foundation}) is called async, if mount the component and unmount immediately without the state been updated yet, componentWillUnmount throws the exception

index.js:896 Uncaught TypeError: Cannot read property 'destroy' of null
    at ChipSet.componentWillUnmount (index.js:896)
    at callComponentWillUnmountWithTimer (react-dom.development.js:16816)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at safelyCallComponentWillUnmount (react-dom.development.js:16823)
    at commitUnmount (react-dom.development.js:17252)
    at unmountHostComponents (react-dom.development.js:17596)
    at commitDeletion (react-dom.development.js:17632)
    at commitAllHostEffects (react-dom.development.js:18462)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at commitRoot (react-dom.development.js:18679)
    at completeRoot (react-dom.development.js:20247)
    at performWorkOnRoot (react-dom.development.js:20170)
    at performWork (react-dom.development.js:20075)
    at performSyncWork (react-dom.development.js:20049)
    at interactiveUpdates$1 (react-dom.development.js:20337)
    at interactiveUpdates (react-dom.development.js:2267)
    at dispatchInteractiveEvent (react-dom.development.js:5083)

**Comment by moog16** [comment]
sounds lke we should have an isMounted_ flag similar to linear progress https://github.com/material-components/material-components-web-react/blob/rc0.10.0/packages/linear-progress/index.tsx#L47.

**Comment by Markus-Schwer** [comment]
I have the same problem with the TextField component.

**Comment by moog16** [comment]
Yes...it looks like this is also an issue there too:

```
  componentWillUnmount() {
    this.state.foundation!.destroy();
  }
```

**Comment by Markus-Schwer** [comment]
This seems to work:
`this.state.foundation && this.state.foundation.destroy()`

**Comment by moog16** [comment]
Ya, would you like to open a PR for that?

**Comment by Markus-Schwer** [comment]
Shure

**Comment by moog16** [comment]
fixed with #870
